persistence/relational-jdbc/build.gradle.kts

```diff
@@ -34,6 +34,7 @@ dependencies {
   compileOnly(libs.jakarta.inject.api)
 
   implementation(libs.smallrye.common.annotation) // @Identifier
+  implementation(libs.postgresql)
```

Comment:
This is creating a hard dependency from this project to the Postgres JDBC driver, FYI.

Reply:
@singhpk234 's reply here: https://github.com/apache/polaris/pull/1802#discussion_r2124954152